File: reports/_workflow/probe-2110-history-live.py
import json
import urllib.parse
import urllib.request

BASE = "http://127.0.0.1:50043"
TOKEN = "demo"


def post(url, data):
    body = urllib.parse.urlencode(data).encode()
    req = urllib.request.Request(url, data=body, method="POST")
    with urllib.request.urlopen(req, timeout=15) as r:
        return r.status, json.loads(r.read().decode())


print("HEALTH")
with urllib.request.urlopen(BASE + "/health", timeout=5) as r:
    print(r.status, r.read().decode()[:500])

print("\nGOLD")
st, gold = post(
    BASE + "/cp/goldgame/single_game_user_gold_history",
    {"token": TOKEN, "gid": "2110"},
)
print("status", st)
print(json.dumps(gold, ensure_ascii=False)[:2500])

print("\nDAY")
day = (gold.get("data", {}).get("list") or [{}])[0].get("day", "")
st, hist = post(
    BASE + "/cp/goldgame/single_game_user_history",
    {"token": TOKEN, "gid": "2110", "day": str(day), "page": "1", "page_size": "30"},
)
print("status", st)
data = hist.get("data") or {}
lst = data.get("list") or []
print("list_len", len(lst), "stats", data.get("statistics"))
if not lst:
    print("EMPTY_LIST")
    print(json.dumps(hist, ensure_ascii=False)[:2000])
else:
    row = lst[0]
    print("parent_keys", sorted(row.keys()))
    print("extend", row.get("extend"))
    print("spe_pos", row.get("spe_pos"), type(row.get("spe_pos")).__name__)
    result = row.get("result")
    print(
        "result_type",
        type(result).__name__,
        "result_keys",
        list(result.keys()) if isinstance(result, dict) else result,
    )
    print("results_len", len(row.get("results") or []))
    if row.get("results"):
        s0 = row["results"][0]
        print("step0_keys", sorted(s0.keys()))
        print("step0.extend", s0.get("extend"))
        print("step0.spe_pos", s0.get("spe_pos"), type(s0.get("spe_pos")).__name__)
        r0 = s0.get("result")
        print(
            "step0.result",
            type(r0).__name__,
            list(r0.keys()) if isinstance(r0, dict) else r0,
        )
        print("step0.frees", s0.get("frees"))
        print("step0.type", s0.get("type"))
        print(
            "parent.bet",
            row.get("bet"),
            "level",
            row.get("level"),
            "end_gold",
            row.get("end_gold"),
            "order_id",
            row.get("order_id"),
        )
    print("FULL_ROW")
    print(json.dumps(row, ensure_ascii=False)[:5000])

File: reports/_workflow/probe-2110-history-live.mjs
const BASE = "http://127.0.0.1:50043";
const TOKEN = "demo";

async function post(url, data) {
  const body = new URLSearchParams(data).toString();
  const res = await fetch(url, {
    method: "POST",
    headers: { "Content-Type": "application/x-www-form-urlencoded" },
    body,
  });
  const text = await res.text();
  let json;
  try {
    json = JSON.parse(text);
  } catch {
    json = { _raw: text.slice(0, 500) };
  }
  return { status: res.status, json };
}

const health = await fetch(BASE + "/health");
console.log("HEALTH", health.status, await health.text());

const gold = await post(BASE + "/cp/goldgame/single_game_user_gold_history", {
  token: TOKEN,
  gid: "2110",
});
console.log("GOLD", gold.status);
console.log(JSON.stringify(gold.json, null, 2).slice(0, 2500));

const day = gold.json?.data?.list?.[0]?.day ?? "";
const hist = await post(BASE + "/cp/goldgame/single_game_user_history", {
  token: TOKEN,
  gid: "2110",
  day: String(day),
  page: "1",
  page_size: "30",
});
console.log("DAY", hist.status);
const data = hist.json?.data || {};
const lst = data.list || [];
console.log("list_len", lst.length, "stats", data.statistics);
if (!lst.length) {
  console.log("EMPTY_LIST");
  console.log(JSON.stringify(hist.json).slice(0, 2000));
} else {
  const row = lst[0];
  console.log("parent_keys", Object.keys(row).sort());
  console.log("extend", row.extend);
  console.log("spe_pos", row.spe_pos, Array.isArray(row.spe_pos));
  console.log(
    "result_type",
    typeof row.result,
    row.result && typeof row.result === "object" ? Object.keys(row.result) : row.result
  );
  console.log("results_len", (row.results || []).length);
  if (row.results?.[0]) {
    const s0 = row.results[0];
    console.log("step0_keys", Object.keys(s0).sort());
    console.log("step0.extend", s0.extend);
    console.log("step0.spe_pos", s0.spe_pos, Array.isArray(s0.spe_pos));
    console.log(
      "step0.result",
      typeof s0.result,
      s0.result && typeof s0.result === "object" ? Object.keys(s0.result) : s0.result
    );
    console.log("step0.frees", s0.frees);
    console.log("step0.type", s0.type);
    console.log("parent.bet", row.bet, "level", row.level, "end_gold", row.end_gold, "order_id", row.order_id);
  }
  console.log("FULL_ROW");
  console.log(JSON.stringify(row).slice(0, 5000));
}
